src/metabase/models/model_index.clj

```diff
@@ -158,6 +159,7 @@
                        :state      (if (> (count values-to-index) max-indexed-values)
                                      "overflow"
                                      "indexed")}))
+        (run! search/update! (t2/select ModelIndexValue :model_index_id (:id model-index)))
```

Comment:
should this be in `future`?.. :\ I mean it's fetching the data, so it's not fantastic to do right here :(

Reply:
This method is already doing a doseq of inserts, I think it's fine 😆

Reply:
Do we need to worry about `t2/select` head holding? Maybe this should be a reducible select.